### ISSUE TITLE
Not able to change programming language once extension is loaded

### ISSUE THREAD

**Issue by vineethreddy899** [piyushgarg-dev/leetcode-whisper-chrome-extension#99]
Not able to change programming language once extension is loaded. The dropdown to select programming language is not working.

**Comment by sanvi1803** [comment]
Hey any idea how to get API Key since for me it says **The model `gpt-4o` does not exist or you do not have access to it.**

**Comment by Owaiseimdad** [comment]
> Not able to change programming language once extension is loaded. The dropdown to select programming language is not working.

@vineethreddy899  can add a video if possible. This should not be the case as I have seen the code of the content.tsx and there os nothing we are doing to language button.

**Comment by piyushgarg-dev** [comment]
Yes, This issue is coming that language dropdown is not working.

**Comment by piyushgarg-dev** [comment]
@Owaiseimdad Done

**Comment by Owaiseimdad** [comment]
@vineethreddy899 let me know what am I doing wrong. Because I cannot reproduce the issue. 

https://github.com/user-attachments/assets/e77f6fd5-04c2-4483-ac68-2390820fc88b

**Comment by ruru-m07** [comment]
@Owaiseimdad are u running build output or are you in dev mode?

also resolve the configuration issue that occurs after the CSS for the next pages loads.

and this issue only happens in the dev environment, not in the prod build.

**Comment by Owaiseimdad** [comment]
got it @ruru-m07  will do that. It was actually prod mode.

**Comment by Owaiseimdad** [comment]
https://github.com/user-attachments/assets/1d1a2d08-b5d6-4af1-a9a9-fbddd8b0e7ef

**Comment by ruru-m07** [comment]
### I did extensive debugging and found a single property causing this issue.

#### I spent a lot of time and effort debugging this issue; I hope this is helpful to all.

# my report | TL;DR

#### so what I do?
- I have faced this kind of issue before, so I know that this conflict arises because our site and the LeetCode site both use Tailwind CSS.
- Instead of using our CSS directly from `/src/index.css`, I manually build Tailwind CSS. For instance, I use `npx tailwindcss -i ./src/index.css -o ./out.css`, where `-i` specifies the input CSS file, and `-o` specifies the output location.
- than use that `out.css` file in project:- 
- as u can see here 

![image](https://github.com/user-attachments/assets/01a675f7-161f-4db8-aeef-17cd81b23a8f)

- this is how CSS build looks like. it was vary large file containing **1844** lines of CSS code.
 
![image](https://github.com/user-attachments/assets/d5e0d6dc-f348-48cb-aaf9-23dceed13245)

- and then I comment each and every Selector and see if the output changes.

- and guess what? I want that single selector that created this issue, it's `.hidden`. If I comment on it then I see that issue resolved

![image](https://github.com/user-attachments/assets/3988c39e-7924-4545-9af4-14d31793a54e)

- it means the selector `.hidden` makes this issue happen.

- but there is no why to tell tailwind that don't make .hidden class also we can't overwrite it.

- so that I write some scripts to handle this. this will run after `npx tailwindcss -i ./src/index.css -o ./out.css --watch` runs
  
![image](https://github.com/user-attachments/assets/c1552b0c-318e-4e21-9e2d-152ad35c7416)

- also here is **run.js** and **package.json**. this is the workflow for development.

![image](https://github.com/user-attachments/assets/2f671c06-47a9-4d31-95e7-b00cafd09571)

- also we not gonna push that large file in **prod** or **github**

![image](https://github.com/user-attachments/assets/ef0a4351-89df-4f71-8ee1-f0355a720c50)

Here is a video proof. ( working selector )

https://github.com/user-attachments/assets/142840d4-9756-4c43-8218-9439740ade8b

here is some codebase walkthrough


https://github.com/user-attachments/assets/824bc92a-2d9f-4da6-b99b-9b135fc173ca

mentions: @Owaiseimdad @piyushgarg-dev @vineethreddy899

**Comment by solaimanTier5** [comment]
Have you considered the possibility of using Shadow Root render rather than direct rendering in content.tsx?

Shadow Root creates an isolated environment for CSS styling and prevents conflicts between the injected extension UI and the host webpage (Leetcode) CSS.

here is an example.

```
import { createRoot } from 'react-dom/client'
import { StrictMode } from 'react'
import ContentPage from '@/content/content'

const container = document.createElement('div')
container.id = '__leetcode_ai_whisper_container'
document.body.append(container)

const shadowRoot = container.attachShadow({ mode: 'open' })

const styleLink = document.createElement('link')
styleLink.rel = 'stylesheet'
styleLink.href = chrome.runtime.getURL('index.css')
shadowRoot.appendChild(styleLink)

const shadowRootContainer = document.createElement('div')
shadowRoot.appendChild(shadowRootContainer)

createRoot(shadowRootContainer).render(
  <StrictMode>
    <ContentPage />
  </StrictMode>
)

```
mentions: @Owaiseimdad @piyushgarg-dev @vineethreddy899
